File: macroslib/tests/expectations/result_in_callback.cpp
r##"class Foo {
public:
    virtual ~Foo() noexcept {}

    virtual std::variant<std::string_view, Error> unpack(std::string_view x) const noexcept = 0;

    virtual std::optional<Error> remove() const noexcept = 0;"##;

r#"union CRustVoidOkResultUnion4232mut3232c_void {
    uint8_t ok;
    void * err;
};"#;

File: macroslib/tests/expectations/result_in_callback.rs
foreigner_class!(class Error {
    self_type Error;
    private constructor = empty;
});

foreign_callback!(
    interface Foo {
        self_type Foo;
        unpack = Foo::unpack(&self, x: &str) -> Result<String, Error>;
        remove = Foo::remove(&self) -> Result<(), Error>;
    }
);
